### PR TITLE
Wasm build improvements

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -277,7 +278,8 @@
     </ReadLinesFromFile>
 
     <PropertyGroup>
-      <EmccFlags>$(_DefaultEmccFlags) $(EmccFlags)</EmccFlags>
+      <EmccFlags Condition="'$(_WasmDevel)' == 'true'">-O0</EmccFlags>
+      <EmccFlags>$(EmccFlags) $(_DefaultEmccFlags) $(EmccFlags)</EmccFlags>
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.targets:282]: This overwrites any previous set value for `EmccFlags`. Instead, if you want to prepend, then:

```xml
<EmccFlags>$(_DefaultEmccFlags) $(EmccFlags)</EmccFlags>
<EmccFlags Condition="'$(_WasmDevel)' == 'true'">-O0 $(EmccFlags)</EmccFlags>
```

File: src/mono/wasm/build/WasmApp.targets [hunk 2]
```diff
@@ -316,8 +318,6 @@
      <_WasmRuntimePackNativeLibs Include="libmono-profiler-aot.a"/>
      <_WasmRuntimePackNativeLibs Include="libicuuc.a" />
      <_WasmRuntimePackNativeLibs Include="libicui18n.a" />
-     <_WasmObjects Include="@(_WasmRuntimePackNativeLibs->'$(MicrosoftNetCoreAppRuntimePackRidDir)\native\%(FileName)%(Extension)')" />
-
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.targets:320]: Why was this removed?

Comment (vargaz) [reply src/mono/wasm/build/WasmApp.targets:320]: The same files were added a few lines later.

Comment (radical) [reply src/mono/wasm/build/WasmApp.targets:320]: 🤦‍♂️

File: src/mono/mono/mini/llvm-runtime.cpp [hunk 3]
```diff
@@ -3,25 +3,13 @@
...
 	gint32 *ex = NULL;
 
-#ifdef HOST_WASM
-		EM_ASM(
-			   var err = new Error();
-			   console.log ("Throw stacktrace: \n");
-			   console.log (err.stack);
-			   );
-#endif
-
```
Comment (radical) [inline src/mono/mono/mini/llvm-runtime.cpp:24]: What are these changes for?

File: src/mono/wasm/Makefile [hunk 4]
```diff
@@ -117,7 +117,7 @@ $(BUILDS_OBJ_DIR)/corebindings.o: runtime/corebindings.c | $(BUILDS_OBJ_DIR)
 	$(EMCC) $(EMCC_FLAGS) $(1) -Oz -I$(MONO_INCLUDE_DIR) runtime/corebindings.c -c -o $$@
 
 $(NATIVE_BIN_DIR)/src/emcc-flags.txt: | $(NATIVE_BIN_DIR)/src Makefile
-	echo "$(call escape_quote,$(EMCC_FLAGS)) $(1) -Oz" > $$@
+	echo "$(call escape_quote,$(EMCC_FLAGS)) $(1)" > $$@
```
Comment (radical) [inline src/mono/wasm/Makefile:120]: Can we add this, and the other changes to the PR description? Maybe like:

```
- Always set `$(WasmNativeStrip)` to `false`, for library tests
- Remove `-Oz` from default emcc flags
- New `$(_WasmDevel)` to build native files with `-O0`
```

Or corrected, if I have some of that wrong!

File: src/mono/mono/mini/llvm-runtime.cpp (hunk 3 above)
Comment (vargaz) [reply src/mono/mono/mini/llvm-runtime.cpp:24]: They were added accidently by 997dc08f691ee0246990a21c73850a041b80b800.

File: src/mono/wasm/Makefile (hunk 4 above)
Comment (vargaz) [reply src/mono/wasm/Makefile:120]: They are in the commit descriptions, will be in the final merged commit message.

Comment (radical) [reply src/mono/wasm/Makefile:120]: Sorry, I missed all those!

**Review by radical — APPROVED** [review]
LGTM

**Comment by vargaz** [comment]
Failures are unrelated.
